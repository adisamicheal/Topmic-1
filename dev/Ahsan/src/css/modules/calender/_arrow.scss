.arrowRight{
		@include arrow(92.4%,67%,40px,40px,135deg ) //Prameters are left,top,height,width,rotationDegree
		border-left: 3px solid $arrowColor;
		border-top: 3px solid $arrowColor;
		transition:all 0.4s ease-in-out;
		&:active{
			transform:scale(1.2) rotate(135deg)
		}
	}
	.arrowLeft{
		@include arrow(4.5%,67%,40px,40px,-45deg) //Prameters are left,top,height,width,rotationDegree
		border-left: 3px solid $arrowColor;
		border-top: 3px solid $arrowColor;
		transition:all 0.4s ease-in-out;
		
		&:active{
			transform:scale(1.2) rotate(-45deg)
		}

	}