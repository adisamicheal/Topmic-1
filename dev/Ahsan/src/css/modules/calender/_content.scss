.heroImage{
	@include heroImage('collage photos.jpg',100%,510px); //parameters are (url,width,height)
	margin:0px 0px -20px 0px;
	.line{
		position: absolute;
		width: 77%;
		height: 3px;
		background-color: #134A7D;
		left: 11.8%;
		top: 71%;
	}
	h2{
		position: absolute;
		top: 20px;
		left: 43%;
		font-family: Roboto;
	}
	
	
}