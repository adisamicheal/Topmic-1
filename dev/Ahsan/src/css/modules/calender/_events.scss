.renderEvents{
	width: 80%;
	height: 100%;
	position: absolute;
	top: 45.5%;
	left: 10.4%;
	.event{
		float:left;
		margin-left: 5%;
		width: 20%;
		height: 25%;
		margin-top: 6%;
		//background-color: red;
		.comingEvents{
			float : left;
			font-weight: bold;
			color: #0078AD;
			font-size:18px ;
			text-align: center;
			margin:-150px auto auto -6%;
			text-transform: uppercase;
			width: 80%;
				.clear{
					clear: both;
				}
			}
		}	
		.comingEventsDates{
				@include event(23px,0px auto auto 5%,28px,130px,130px,#134A7D,36px)// Parameters are fontsize,margin,lineHeight,width,height,color,padding
				background-color: #fff;
				float: left;
				border-radius: 100%;
				font-weight: bold;
				text-align: center;
				box-shadow: 0px 4px 2px rgba(238, 155, 37,0.7);
				text-transform: uppercase;
		}

		.time{
			float:left;
			text-align: center;
			width: 140px;
			height: 140px;
			font-weight: bold;
			color: #134A7D;
			margin:25px auto auto 15px;
			font-size: 18px;
		}
}
