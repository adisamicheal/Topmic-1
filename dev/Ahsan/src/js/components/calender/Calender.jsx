import React from 'react';
import Content from './Content';
import events from './Data'

export default class Calender extends React.Component {
	constructor(){
		super();
		
		this.state={
			n:0
		}
		this.onClickRightArrow=this.onClickRightArrow.bind(this);
		this.onClickLeftArrow=this.onClickLeftArrow.bind(this);
	}

	onClickLeftArrow(){
		let {n}=this.state
		if(n<=0){
			return this.setState({
				n:0
			})
		}
		else{
			return this.setState({
				n:n-1
			})
		}	
	}
	onClickRightArrow(){
		let {n}=this.state
		if(this.state.n >= events.length-4){
			return this.setState({
				n:events.length-4
			})
		}
		else{
			return this.setState({
				n:n+1
			})
		}	
	}
	render(){
		const {n}=this.state;
		return (
			<div>
			<Content
			 n={n}
			 onClickLeftArrow={this.onClickLeftArrow}
			 onClickRightArrow={this.onClickRightArrow}
			 />	
			</div>
		)
	}
} 