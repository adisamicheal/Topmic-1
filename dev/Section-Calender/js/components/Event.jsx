import React from 'react'
import events from '../Data'


const Event=({n})=>{
	let selectedEvent=[...events];
		return selectedEvent.slice(n,n+4).map(({...item},index)=>(

				<div className="event">
					<div className="comingEvents">
						{item['event']}
					</div>
					<div className="comingEventsDates">
							{item['eventDate']}
					</div>
					<div className="time">
							{item['time']}
					</div>
				</div>
				

		))
	}
export default Event;