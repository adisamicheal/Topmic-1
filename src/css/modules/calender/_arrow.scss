.arrowRight {
	@include arrow(92.4%, 67%, 40px, 40px, 135deg);
	border-left: 3px solid $arrowColor;
	border-top: 3px solid $arrowColor;
	transition: all 0.4s ease-in-out;
	@media (max-width: 767px) {
		left: 90.6%;
	}
	&:active {
		transform: scale(1.5) rotate(135deg);
		outline: 0;
		outline: none;
	}
	&:focus {
		border-left: 6px solid $arrowColor;
		border-top: 6px solid $arrowColor;
		outline: 0;
		outline: none;
	}
}
.arrowLeft {
	@include arrow(4.5%, 67%, 40px, 40px, -45deg);
	border-left: 3px solid $arrowColor;
	border-top: 3px solid $arrowColor;
	transition: all 0.4s ease-in-out;
	&:active {
		transform: scale(1.5) rotate(-45deg);
		outline: 0;
		outline: none;
	}
	&:focus {
		border-left: 6px solid $arrowColor;
		border-top: 6px solid $arrowColor;
		outline: 0;
		outline: none;
	}
}






@media (max-width: 526px) {
	.arrowRight {
	@include arrow(85.4%, 67%, 40px, 40px, 135deg);
	border-left: 3px solid $arrowColor;
	border-top: 3px solid $arrowColor;
	transition: all 0.4s ease-in-out;
	
	&:active {
		transform: scale(1.5) rotate(135deg);
		outline: 0;
		outline: none;
	}
	&:focus {
		border-left: 6px solid $arrowColor;
		border-top: 6px solid $arrowColor;
		outline: 0;
		outline: none;
	}
}
.arrowLeft {
	@include arrow(4.5%, 67%, 40px, 40px, -45deg);
	border-left: 3px solid $arrowColor;
	border-top: 3px solid $arrowColor;
	transition: all 0.4s ease-in-out;
	&:active {
		transform: scale(1.5) rotate(-45deg);
		outline: 0;
		outline: none;
	}
	&:focus {
		border-left: 6px solid $arrowColor;
		border-top: 6px solid $arrowColor;
		outline: 0;
		outline: none;
	}
}

}