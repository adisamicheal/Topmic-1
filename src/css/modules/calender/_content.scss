.heroImage {
	@include heroImage(
		'collage photos.jpg',
		100%,
		510px
	); //parameters are (url,width,height)
	margin: 0px 0px -20px 0px;
	.line {
		position: absolute;
		width: 77%;
		height: 3px;
		background-color: #134a7d;
		left: 11.8%;
		top: 71%;
		@media (max-width: 767px) {
			width: 80%;
			left: 10.7%;
		}
	}
	h2 {
		position: absolute;
		top: 20px;
		left: 43%;
		font-family: Roboto;
		@media (max-width: 767px) {
			position: absolute;
			left: 37%;
			font-size: 25px;
		}
	}
}

@media (max-width: 600px) {
	.heroImage {
		@include heroImage(
			'collage photos.jpg',
			100%,
			510px
		); //parameters are (url,width,height)
		margin: 0px 0px -20px 0px;
		.line {
			position: absolute;
			width: 77%;
			height: 3px;
			background-color: #134a7d;
			left: 11.8%;
			top: 71%;
		}
		h2 {
			position: absolute;
			top: 20px;
			left: 29%;
			font-family: Roboto;
		}
	}
}

@media (max-width: 468px) {
	.heroImage {
		@include heroImage(
			'collage photos.jpg',
			100%,
			510px
		); //parameters are (url,width,height)
		margin: 0px 0px -20px 0px;
		.line {
			position: absolute;
			width: 77%;
			height: 3px;
			background-color: #134a7d;
			left: 11.8%;
			top: 71%;
		}
		h2 {
			position: absolute;
			top: 30px;
			left: 20%;
			font-family: Roboto;
		}
	}
}
