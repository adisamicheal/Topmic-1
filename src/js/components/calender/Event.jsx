import React from 'react';
import { connect } from 'react-redux';
import { arrayOf, shape, number, string, func} from 'prop-types';
import * as actions from '../../actions'

export class Event extends React.Component {

	componentWillMount(){
		this.props.getCalenderData();
	}
	renderArray() {
		const {count,calender} = this.props;
		let events;
		const {counter}=calender;
		if(Object.keys(this.props.calender).length){
			 events=[...calender.events]
			 // counter=calender.counter;
		}
		else{
			return '';
		}
		return events
			.slice(counter, counter + count)
			.map(({ ...item }) => (
				<div key={item.id} className="event">
					<div className="comingEvents">{item.event}</div>
					<div className="comingEventsDates">{item.eventDate}</div>
					<div className="time">{item.time}</div>
					<div className="clear" />
				</div>
			));
	}
	render() {
		return <div>{this.renderArray()}</div>;
	}
}

function mapStateToProps({ calender }) {
	return {
		calender
	};
}

Event.propTypes = {
	count: number.isRequired,
	calender:shape({
		counter: number.isRequired,
		events: arrayOf(
			shape({
				id: number.isRequired,
				event: string.isRequired,
				eventDate: string.isRequired,
				time: string.isRequired
			})
		).isRequired
	}).isRequired,
	getCalenderData:func.isRequired
	
};
export default connect(mapStateToProps, {getCalenderData:actions.getCalenderData})(Event);
