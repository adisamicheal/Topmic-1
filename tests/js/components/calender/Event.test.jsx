import React from 'react';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { Event } from '../../../../src/js/components/calender/Event';

Enzyme.configure({ adapter: new Adapter() });

describe('Test the Event Component of Calender', () => {
	let calender,events;
	calender={
		counter:0,
		events :[
			{
				id: 1,
				event: 'Half terms End',
				eventDate: '01 May',
				time: ''
			},
			{
				id: 2,
				event: 'NetBall: first vii vs viii',
				eventDate: '20 May',
				time: '2:30 pm'
			},
			{
				id: 3,
				event: 'Parent presentation on healthy living',
				eventDate: '03 June',
				time: '12:35 pm'
			},
			{
				id: 4,
				event: 'M5 rs trip to walsigham',
				eventDate: '25 June',
				time: ''
			},
			{
				id: 5,
				event: 'Tour to muree',
				eventDate: '15 July',
				time: '2:30 pm'
			},
			{
				id: 6,
				event: 'Water splash Day',
				eventDate: '28 July',
				time: '11:35 am'
			},
			{
				id: 7,
				event: 'Visit to Jungle',
				eventDate: '05 Aug',
				time: ''
			},
			{
				id: 8,
				event: 'Independence Day',
				eventDate: '14 Aug',
				time: '12:35 am'
			},
			{
				id: 9,
				event: 'Bsant Day',
				eventDate: '25 Aug',
				time: '11:38 pm'
			}
		]			
	}
	events =
		[
			{
				id: 1,
				event: 'Half terms End',
				eventDate: '01 May',
				time: ''
			},
			{
				id: 2,
				event: 'NetBall: first vii vs viii',
				eventDate: '20 May',
				time: '2:30 pm'
			},
			{
				id: 3,
				event: 'Parent presentation on healthy living',
				eventDate: '03 June',
				time: '12:35 pm'
			},
			{
				id: 4,
				event: 'M5 rs trip to walsigham',
				eventDate: '25 June',
				time: ''
			},
			{
				id: 5,
				event: 'Tour to muree',
				eventDate: '15 July',
				time: '2:30 pm'
			},
			{
				id: 6,
				event: 'Water splash Day',
				eventDate: '28 July',
				time: '11:35 am'
			},
			{
				id: 7,
				event: 'Visit to Jungle',
				eventDate: '05 Aug',
				time: ''
			},
			{
				id: 8,
				event: 'Independence Day',
				eventDate: '14 Aug',
				time: '12:35 am'
			},
			{
				id: 9,
				event: 'Bsant Day',
				eventDate: '25 Aug',
				time: '11:38 pm'
			}
		]			
	const count = 0;
	const getCalenderData=jest.fn()
	const wrapper = shallow(
		<Event calender={calender} getCalenderData={getCalenderData} />
	);
	it('checks the rendring', () => {
		expect(wrapper).toMatchSnapshot();
	});
	it('checks the array', () => {
		expect(1).toBe(events[0].id);
	});
	it('checks the render function', () => {
		const renderArray = jest.fn();
		expect(renderArray.mock.calls.length).toBe(0);
	});
	it('test the getCalenderData function',()=>{
		expect(getCalenderData.mock.calls.length).toBe(1)
	})
	it('test the Object.keys',()=>{
		expect(Object.keys(calender).length).toBe(2);
	})
	it('test the Object.keys.counter',()=>{
		expect(calender.counter).toBe(0);
	})
	it('test the Object.keys.counter',()=>{
		calender=''
		expect(calender).toBe('');
	})
});
