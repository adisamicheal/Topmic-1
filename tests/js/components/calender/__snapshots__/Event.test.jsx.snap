// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test the Event Component of Calender checks the rendring 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Event
    calender={
      Object {
        "counter": 0,
        "events": Array [
          Object {
            "event": "Half terms End",
            "eventDate": "01 May",
            "id": 1,
            "time": "",
          },
          Object {
            "event": "NetBall: first vii vs viii",
            "eventDate": "20 May",
            "id": 2,
            "time": "2:30 pm",
          },
          Object {
            "event": "Parent presentation on healthy living",
            "eventDate": "03 June",
            "id": 3,
            "time": "12:35 pm",
          },
          Object {
            "event": "M5 rs trip to walsigham",
            "eventDate": "25 June",
            "id": 4,
            "time": "",
          },
          Object {
            "event": "Tour to muree",
            "eventDate": "15 July",
            "id": 5,
            "time": "2:30 pm",
          },
          Object {
            "event": "Water splash Day",
            "eventDate": "28 July",
            "id": 6,
            "time": "11:35 am",
          },
          Object {
            "event": "Visit to Jungle",
            "eventDate": "05 Aug",
            "id": 7,
            "time": "",
          },
          Object {
            "event": "Independence Day",
            "eventDate": "14 Aug",
            "id": 8,
            "time": "12:35 am",
          },
          Object {
            "event": "Bsant Day",
            "eventDate": "25 Aug",
            "id": 9,
            "time": "11:38 pm",
          },
        ],
      }
    }
    getCalenderData={
      [MockFunction] {
        "calls": Array [
          Array [],
        ],
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [],
    },
    "ref": null,
    "rendered": Array [],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [],
      },
      "ref": null,
      "rendered": Array [],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
